pkg/deploy-manager/subscription.go

```diff
@@ -360,3 +360,95 @@ func (t *DeployManager) waitForOCSOperator() error {
 
 	return nil
 }
+
+// UninstallOCS uninstalls ocs operator and storage clusters
+func (t *DeployManager) UninstallOCS(ocsRegistryImage string, localStorageRegistryImage string) error {
+	err := t.deleteNoobaaSystemsAndWait()
+	if err != nil {
+		return err
+	}
+
+	err = t.deleteStorageClusters()
+	if err != nil {
+		return err
+	}
+
```

Comment:
this logic is a bit of a moving target right now. Sorry about that.

The noobaa system moved to the StorageCluster object and is treated differently now that it was previously. Now noobaa is re-created by the storagecluster if it gets deleted. Previously it was possible to delete noobaa without it getting immediately re-created, but that's no longer the case.

It's actually probably going to be easier to address this in the reconcile loop by adding a finalizer to the storagecluster object that ensures the noobaa object is removed first before tearing down the cephcluster or anything else. The alternative would be to orphan delete the StorageCluster object and clean up each object in deploy-manager, but that seems like more work.

Reply:
Added a finalizer in the reconcile loop and deleted noobaa (with pvc) in the predelete hook. So deleting storagecluster will cause deletion of noobaa object first, then ceph and rest of the resourced will be deleted once finalizer has been removed.